tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -0,0 +1,1359 @@
+from pyccel.decorators import types
+from pyccel import epyccel
+import pytest
+from numpy.random import rand, randint
+from numpy import isclose
+
+# Functions still to be tested:
+#    full_like
+#    empty_like
+#    zeros_like
+#    ones_like
+#    array
+#    # ...
+#    norm
+#    int
+#    real
+#    imag
+#    float
+#    double
+#    mod
+#    float32
+#    float64
+#    int32
+#    int64
+#    complex128
+#    complex64
+#    matmul
+#    sum
+#    prod
+#    product
+#    linspace
+#    diag
+#    where
+#    cross
+#    # ---
+
+def test_fabs_call():
+    @types('real')
+    def fabs_call(x):
+        from numpy import fabs
+        return fabs(x)
+
+    f1 = epyccel(fabs_call)
+    x = rand()
+    assert(isclose(f1(x), fabs_call(x), rtol=1e-15, atol=1e-15))
+
+def test_fabs_phrase():
+    @types('real','real')
+    def fabs_phrase(x,y):
+        from numpy import fabs
+        a = fabs(x)*fabs(y)
+        return a
+
+    f2 = epyccel(fabs_phrase)
+    x = rand()
+    y = rand()
+    assert(isclose(f2(x,y), fabs_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+@pytest.mark.xfail
+def test_fabs_return_type():
+    @types('int')
+    def fabs_return_type(x):
+        from numpy import fabs
+        a = fabs(x)
+        return a
+
+    f1 = epyccel(fabs_return_type)
+    x = randint(100)
+    assert(isclose(f1(x), fabs_return_type(x), rtol=1e-15, atol=1e-15))
+    assert(type(f1(x)) == type(fabs_return_type(x))) # pylint: disable=unidiomatic-typecheck
+
+def test_absolute_call():
+    @types('real')
+    def absolute_call(x):
+        from numpy import absolute
+        return absolute(x)
+
+    f1 = epyccel(absolute_call)
+    x = rand()
+    assert(isclose(f1(x), absolute_call(x), rtol=1e-15, atol=1e-15))
+
+def test_absolute_phrase():
+    @types('real','real')
+    def absolute_phrase(x,y):
+        from numpy import absolute
+        a = absolute(x)*absolute(y)
+        return a
+
+    f2 = epyccel(absolute_phrase)
+    x = rand()
+    y = rand()
+    assert(isclose(f2(x,y), absolute_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+@pytest.mark.xfail
```

Comment:
Could you write the reason why this test fails?

Reply:
Sorry this test shouldn't fail. It was failing because numpy returns numpy types rather than python types. It is therefore necessary to use `.item()` in order to get the underlying class